Added DAI for us and pspeu

ProjectOblivion opened the pull request:
All symbols, text segments, and rodata segments were automatically generated.

ProjectOblivion commented on src/st/dai/blit_char.c:
```diff
@@ -0,0 +1,6 @@
+
+INCLUDE_ASM("st/dai/nonmatchings/blit_char", BlitChar);
+
+INCLUDE_RODATA("st/dai/nonmatchings/blit_char", D_us_801C0080);
```
@Xeeynamo @sozud @bismurphy 
Do you have an opinion 👍 or 👎 on this?  If it makes sense to everyone else, I'd like to adjust it for this PR.

sozud replied:
Bismurphy just did BlitChar for no3 a couple of days ago, let's do what he thinks? https://github.com/Xeeynamo/sotn-decomp/blob/8b15d522b79a834889d3997bb3480563a5a830ea/src/st/no3/blit_char.c#L5

ProjectOblivion replied:
Not arguing the response, but asking for clarity.
There are several different implementations between different overlays and us vs pspeu.  Even the same overlays don't have the same implementation between us and pspeu.  What would I match?

ProjectOblivion commented:
You can do this for yourself by cloning https://github.com/ProjectOblivion/decomp_utils into the tools directory and running `python3 tools/decomp_utils/generate_sotn_config.py dai` for each version.  You'll want to use the dev branch because I found a couple issues while doing this that I resolved but haven't pushed to main.

I manually added the clang-format lines.  I'd like to add this to the tool output, though I'm not sure of the specific criteria for when it is needed.

I did some things manually that I've been working on, but I don't have them incorporated functionally into the tool yet.
- e_init.c generation
- header.c generation
- Added DAI_EInitUnkBreakable, g_EInitObtainable, g_EInitParticle, g_EInitSpawner, g_EInitUnkId13, g_EInitLockCamera, g_EInitDamageNum, and DAI_BackgroundBlockInit to the symbols file
- Adding the splat suggestion comments

I've been looking at replacing the INCLUDE_ASM with actual code or include automatically, but I don't think it is a capability that is right for the initial generation.  The places where that was done were done manually, which includes the data and bss segments related to them.

I added the isolated comments manually when I validated the match for ones that didn't have enough context just from the .c file.

I also had to do a few additional things manually for pspeu:
- Added EntityMagicallySealedDoor, DAI_Load, and DAI_Overlay to the symbols file, working on doing it automatically
- Added the 3 GetLang functions to the symbols file and named them manually.  I'm thinking on how to handle the naming automatically in a way that also correctly preserves their association with their respective segments
- DAI is the first with EntityMagicallySealedDoor so I added that manually
- Since the GetLang functions aren't named automatically, e_magically_sealed_door, e_mist_door, and e_cutscene aren't handled correctly automatically and e_misc has a different start than typical.  I got around this by adjusting the start functions for these, but it isn't something I wanted to do permanently.  The fragility of this segment id method makes it more suited as an override, but I have not yet incorporated a replacement.

One known issue is that 0x801C58C4 is misattributed as func_8018F890 instead of func_801961DC.  I have resolution for this, but it is not incorporated yet.

ProjectOblivion commented on src/st/dai/e_init.c:
```diff
@@ -0,0 +1,150 @@
+void EntityMistDoor(Entity* self);
+
+PfnEntityUpdate OVL_EXPORT(EntityUpdates)[] = {
+    /* 0x00 */ EntityUnkBreakable,
+    /* 0x04 */ EntityExplosion,
+    /* 0x08 */ EntityPrizeDrop,
+    /* 0x0C */ EntityDamageDisplay,
+    /* 0x10 */ DAI_EntityRedDoor,
+    /* 0x14 */ EntityIntenseExplosion,
+    /* 0x18 */ EntitySoulStealOrb,
+    /* 0x1C */ EntityRoomForeground,
+    /* 0x20 */ EntityStageNamePopup,
+    /* 0x24 */ EntityEquipItemDrop,
+    /* 0x28 */ EntityRelicOrb,
+    /* 0x2C */ EntityHeartDrop,
```
I think I did what you were getting at, let me know if I missed the mark.
